stl/inc/ranges

```diff
@@ -1548,7 +1548,7 @@ namespace ranges {
             }
 
             // clang-format off
-            template <bool _OtherConst = _Const>
+            template <bool _OtherConst = !_Const>

```

Comment:
Was this intentional?

Reply:
Thanks for checking - yes, this is intentionally following https://eel.is/c++draft/range.take.sentinel#4 . I missed this while reviewing #1457 but noticed it while checking all of the equality operators here.